mypyc/lib-rt/dict_ops.c

```diff
@@ -80,6 +80,34 @@ PyObject *CPyDict_SetDefaultWithNone(PyObject *dict, PyObject *key) {
     return CPyDict_SetDefault(dict, key, Py_None);
 }
 
+PyObject *CPyDict_SetDefaultWithEmptyDatatype(PyObject *dict, PyObject *key,
+                                              int data_type) {
+    PyObject *res = PyDict_GetItemWithError(dict, key);
```

Comment:
`PyDict_` functions usually can't be used directly, since they don't work for all dict subclasses. So it's better to continue to use `CPyDict_GetItem` as before, and only add a comment about why we clear the error.